src/fontinfo.rs

```diff
@@ -5,15 +5,178 @@ use serde::{Deserialize, Serialize};
 use crate::shared_types::Guideline;
 use crate::Error;
 
-// The specification is vague about data type limits, usually implicitly meaning
-// Python types. Since Python is dynamic, the spec does not nail down the exact type
-// in several locations and we have to assume a bigger type that can hold all sane
-// values.
-type Integer = i32;
-type NonNegativeInteger = u32;
-type IntegerOrFloat = f64;
-type Float = f64;
-type Bitlist = Vec<u8>;
+/// Fontinfo value types are put into a module so that code on this level cannot access the
+/// literal constructors.
+mod types {
+    use std::convert::TryFrom;
+    use std::ops::Deref;
+
+    use serde::de::Deserializer;
+    use serde::ser::Serializer;
+    use serde::{Deserialize, Serialize};
+
+    use super::Error;
+
+    pub type Integer = i32;
+    pub type NonNegativeInteger = u32;
+    pub type Float = f64;
+    pub type Bitlist = Vec<u8>;
+
+    /// IntegerOrFloat represents a number that can be an integer or float. It should
+    /// serialize to an integer if it effectively represents one.
+    #[derive(Debug, Clone, Copy, PartialEq)]
+    pub struct IntegerOrFloat(f64);
+
+    impl IntegerOrFloat {
+        pub fn new(value: f64) -> Self {
+            IntegerOrFloat(value)
+        }
+
+        pub fn get(&self) -> f64 {
+            self.0
+        }
+
+        pub fn set(&mut self, value: f64) {
+            self.0 = value
+        }
+
+        pub fn is_integer(&self) -> bool {
+            (self.0 - self.round()).abs() < std::f64::EPSILON
+        }
+    }
+
+    impl Deref for IntegerOrFloat {
```

Comment:
Do you have a specific motivation for this?

Reply:
nope. just read that this gives the newtype the same methods as the inner type which may be what we want?

Reply:
Makes it slightly less to type in some other lines of code :)